src/main/java/ee/bitweb/core/retrofit/logging/RetrofitLoggingInterceptor.java

```diff
@@ -0,0 +1,350 @@
+package ee.bitweb.core.retrofit.logging;
+
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.*;
+import okio.Buffer;
+import okio.GzipSource;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.*;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static okhttp3.internal.http.HttpHeaders.promisesBody;
+
+@Slf4j
+public class RetrofitLoggingInterceptor implements LoggingInterceptor {
```

Comment:
We might want to add some sort of white/blacklist logic to control for which endpoints we want to log/omit bodies.
This would allow for example to omit response bodies for requests that contain confidential data in body and handle that separately.